extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Layout} from '../../../src/layout';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {Services} from '../../../src/services';
+import {SocialShare} from './social-share';
+import {addParamsToUrl} from '../../../src/url';
+import {dict} from '../../../src/utils/object';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {getSocialConfig} from './amp-social-share-config';
+import {isExperimentOn} from '../../../src/experiments';
+import {toggle} from '../../../src/style';
+import {user, userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-social-share';
+
+class AmpSocialShare extends PreactBaseElement {
+  /** @override */
+  init() {
+    const viewer = Services.viewerForDoc(this.element);
+    const platform = Services.platformFor(window);
+    const type = userAssert(
+      this.element.getAttribute('type'),
+      'The type attribute is required. %s',
+      this.element
+    );
+    const typeConfig = this.getTypeConfigOrUndefined_(type, viewer, platform);
+    // Hide/ignore component if typeConfig is undefined
+    if (!typeConfig) {
+      toggle(this.element, false);
+      user().warn(TAG, `Skipping obsolete share button ${type}`);
+      return;
+    }
+    this.setHrefAndTargetContext_(typeConfig, platform);
+    if (this.element.getAttribute('layout') === Layout.RESPONSIVE) {
+      return dict({'size': {}});
+    }
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    userAssert(
+      isExperimentOn(this.win, 'amp-social-share-v2'),
+      'expected amp-social-share-v2 experiment to be enabled'
+    );
+    return true;
+  }
+
+  /**
+   * @private
+   * @param {string} type
+   * @param {!../../../src/service/viewer-interface.ViewerInterface} viewer
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   * @return {!JsonObject|undefined}
+   */
+  getTypeConfigOrUndefined_(type, viewer, platform) {
```

Comment:
If none of this depends on the class state, let's move out as a module function.